flow/project.py

```diff
@@ -2693,32 +3272,102 @@ def _guess_cmd(func, name, **kwargs):
+    def make_group(cls, name, options=None):
+        """Make a FlowGroup named ``name`` and return a decorator to make groups.
+
+        .. code-block:: python
+            example_group = FlowProject.make_group('example')
+
+            @example_group
+            @FlowProject.operation
+            def foo(job):
+                return "hello world"
+
+        FlowGroups group operations together for running and submitting
+        JobOperations.
+
+        :param name:
+            The name of the :class:`~.FlowGroup`.
+        :type name:
+            str
+        :param options:
+            A string to append to submissions can be any valid :meth:`FlowOperation.run` option.
+        :type options:
+            str
+        """
+        # Gets the relative filepath of the function caller
+        if name in cls._GROUP_NAMES:
+            raise ValueError("Repeat definition of group with name '{}'.".format(name))
+        else:
+            cls._GROUP_NAMES.add(name)
+
+        group_entry = FlowGroupEntry(name, options)
+        cls._GROUPS.append(group_entry)
+        return group_entry
+
+    def _register_groups(self):
+        "Register all groups and add the correct operations to each."
+        group_entries = []
+        # Gather all groups from class and parent classes.
+        for cls in type(self).__mro__:
+            group_entries.extend(getattr(cls, '_GROUPS', []))
+
+        # Initialize all groups without operations
+        for entry in group_entries:
+            self._groups[entry.name] = FlowGroup(entry.name, options=entry.options)
+
+        # Add operations and directives to group
+        for (op_name, op) in self._operations.items():
+            if isinstance(op, FlowCmdOperation):
+                func = op._cmd
+            else:
+                func = self._operation_functions[op_name]
+
+            if hasattr(func, '_flow_groups'):
+                operation_directives = getattr(func, '_flow_group_operation_directives', dict())
+                for group_name in func._flow_groups:
+                    self._groups[group_name].add_operation(
+                        op_name, op, operation_directives.get(group_name, None))
+
+            # For singleton groups add directives
+            self._groups[op_name].operation_directives[op_name] = getattr(func,
+                                                                          '_flow_directives',
+                                                                          dict())
+
     @property
     def operations(self):
         "The dictionary of operations that have been added to the workflow."
         return self._operations
 
-    def _eligible_for_submission(self, job_operation):
-        """Determine if a job-operation is eligible for submission.
+    @property
+    def groups(self):
+        return self._groups
+
+    def _eligible_for_submission(self, flow_group, job):
+        """Determine if a flow_group is eligible for submission wrt a given job.
```

Comment:
Avoid abbreviations like "with respect to" that might be unclear to readers.
```suggestion
        """Determine if a flow_group is eligible for submission with a given job.
```